### PR TITLE
868: Skara should ignore .gitconfig and /etc/gitconfig

### PR DESCRIPTION
**Pull request by magicus**
Skara is already ignoring .hgrc for mercurial, but the corresponding solution for git was never put in place.

I also fixed a problem where an error condition would cause an infinite loop, which I ran into during testing.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-868](https://bugs.openjdk.java.net/browse/SKARA-868): Skara should ignore .gitconfig and /etc/gitconfig

### Reviewers
 * [Erik Helin](https://openjdk.java.net/census#ehelin) (@edvbld - **Reviewer**) ⚠️ Review applies to 96e826123adb9dcadd11ca971dc238815e6485db

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/997/head:pull/997`
`$ git checkout pull/997`

### CODE REVIEW

**Comment by edvbld** [comment]
@magicus thanks for contributing! What do you think about adding a unit test to `RepositoryTests.java`?

**Comment by magicus** [comment]
@edvbld What exactly do you propose to test in the unit test? This is really solving a problem with the environment -- the tests fail when I run as my user due to conflicting modifications in my `~/.gitconfig`. I can't really modify the users personal gitconfig file in a unit test. The only thing I could really think if is verifying that `git config -l` is empty. Hm, maybe that's a good idea...

**Comment by edvbld** [comment]
@magicus can't you run a test that asserts that `user.name` is empty? Almost everyone configures `user.name` (Git sort-of requires it for commits). In a CI environment the test will of course be trivially correct since we never setup `user.name`, but the test will be accurate when run on developer's personal machines.

**Comment by magicus** [comment]
If `user.name` is set, then `git config -l` will not be empty. :)

File: process/src/main/java/org/openjdk/skara/process/Process.java [hunk 1]
```diff
@@ -67,6 +67,13 @@ public Description environ(String key, String value) {
             return this;
         }
 
+        public Description environMap(Map<String, String> keyValueMap) {
```
Comment (edvbld) [inline process/src/main/java/org/openjdk/skara/process/Process.java:70]: I would have just overloaded `environ`, as in `public Description environ(Map<String, String> env)`

**Review by edvbld — APPROVED** [review]
Looks good, just a minor comment inline.

**Pull request closed without merging**